ext/tags.go

```diff
@@ -23,25 +23,54 @@ var (
 
 	// SamplingPriority determines the priority of sampling this Span.
 	SamplingPriority = uint16Tag("sampling.priority")
+
+	// SpanKind hints at relationship between spans, e.g. client/server
+	SpanKind = spanKindTag("span.kind")
+
+	// SpanKindRPCClient marks a span representing the client-side of an RPC
+	// or other remote call
+	SpanKindRPCClient = SpanKindEnum("c")
+
+	// SpanKindRPCServer marks a span representing the server-side of an RPC
+	// or other remote call
+	SpanKindRPCServer = SpanKindEnum("s")
 )
 
+// ---
+
+// SpanKindEnum represents common span types
+type SpanKindEnum string
```

Comment:
The plain types here make this pretty confusing to use - why not something like

```go
type SpanKind interface {
  spanKindMarker()
  Set(span opentracing.Span, value string)
}

type spanKindImpl struct {
  name string
}
func (sk *spanKindImpl) spanKindMarker() {}
func (sk *...) Set(...) { ... }

var SpanKindRPCClient SpanKind = (&spanKindImpl{name:"s"})(nil)
```
func (